src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DoNotUseStackallocInLoops.cs

```diff
@@ -41,6 +41,10 @@ public sealed override void Initialize(AnalysisContext context)
                 {
                     switch (node.Kind())
                     {
+                        // Don't warn if a local function containing stackalloc is inside a loop.
+                        case SyntaxKind.LocalFunctionStatement:
```

Comment:
Can we also bail out on OperationKind.AnonymousFunction? Please add a test for it as well.

Reply:
I was thinking about that but without dataflow we cannot guarantee the func won't be used within a loop. I guess it's probably better to miss a few cases rather than wrongly reporting.

Reply:
It actually doesn't matter if the func is used in a loop. The stack memory will be released upon the func's return.

Reply:
@mavasani I've added a check for OperationKind.AnonymousFunction.
Should I also add one for FlowAnonymousFunction? I honestly don't know the difference ;-(

Should I also move the SyntaxKind.LocalFunctionStatement check and use OperationKind.LocalFunction instead? I think it doesn't matter, but just thought to group the checks together instead of making one dependent on SyntaxKind and the other dependent on OperationKind.

Reply:
My bad, I didn’t see the current code was operating on SyntaxKinds rather then operation kinds. Instead of fetching operation, it’s better to just bail out on different lambda related syntax kinds.

Reply:
These are `ParenthesizedLambdaExpression`, `SimpleLambdaExpression`, and `AnonymousMethodExpression` only? or I'm missing any others?

Reply:
Thanks, that seems to match compiler implementation: https://github.com/dotnet/roslyn/blob/9a658fab0803051491fa888bbed5a6668b8874e3/src/Compilers/CSharp/Portable/Syntax/SyntaxNodeExtensions.cs#L19-L31